# **0. COLLECTING DATA**

## Import thư viện cần thiết

In [1]:
import pandas as pd

## Đọc dữ liệu

In [2]:
df = pd.read_csv("../data/vn_music_all.csv")
print("Tổng quan dữ liệu:")
df.info()

Tổng quan dữ liệu:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7353 entries, 0 to 7352
Data columns (total 31 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   video_id                     7353 non-null   object 
 1   published_at                 7353 non-null   object 
 2   channel_id                   7353 non-null   object 
 3   channel_title                7353 non-null   object 
 4   title                        7353 non-null   object 
 5   description                  0 non-null      float64
 6   tags                         0 non-null      float64
 7   category_id                  7353 non-null   int64  
 8   default_language             7353 non-null   object 
 9   default_audio_language       7353 non-null   object 
 10  is_vietnamese_language_flag  7353 non-null   bool   
 11  has_vietnamese_diacritics    7353 non-null   bool   
 12  in_publish_range             7353 non-null   bool   
 13 

In [3]:
df.head(5)

,video_id,published_at,channel_id,channel_title,title,description,tags,category_id,default_language,default_audio_language,...,licensed_content,privacy_status,made_for_kids,tags_count,topic_categories,raw_json,source_folder,source_csv_path,search_query_used,crawl_timestamp_utc
0,zTMI-LgAeyE,2025-12-31T23:01:38Z,UC2CjF8lJcRxRZ1TCsceVP7g,Nhạc Test Loa Chuẩn Nhất,LK Nhạc Xuân Không Lời SÔI ĐỘNG NHẤT2026 - Hòa...,NaN,NaN,10,vi,vi,...,True,public,False,26,https://en.wikipedia.org/wiki/Music|https://en...,"{""kind"": ""youtube#video"", ""etag"": ""5qAVZzfzzn7...",nhạc không lời,/home/doanxem99/programming/school/DataVisuali...,nhạc không lời,2026-04-30T16:39:06Z
1,Xua2LkJ0jM0,2025-12-31T23:01:00Z,UCaLvdyzSFAfrB0A2BNQZYhQ,Relaxing Piano Music,Nhạc Không Lời Tăng Năng Lượng Tích Cực Cho Ng...,NaN,NaN,10,vi,vi,...,False,public,False,19,https://en.wikipedia.org/wiki/Classical_music|...,"{""kind"": ""youtube#video"", ""etag"": ""fFrHMJmKan4...",nhạc không lời,/home/doanxem99/programming/school/DataVisuali...,nhạc không lời,2026-04-30T16:39:06Z
2,6jEYb_2QXAI,2025-12-31T22:16:26Z,UC6dg-5Dg6fB3B7Kj6MK_umw,Tình Khúc Trữ Tình,"Nhạc BOLERO Trữ Tình Hay Nhất, Độc Lạ 2025 - N...",NaN,NaN,10,vi,vi,...,True,public,False,27,https://en.wikipedia.org/wiki/Music|https://en...,"{""kind"": ""youtube#video"", ""etag"": ""cpI-C01KYNC...",nhạc,/home/doanxem99/programming/school/DataVisuali...,nhạc,2026-04-30T15:34:13Z
3,MqjTEJ6NYJg,2025-12-31T12:30:07Z,UCocyTID1JieKnhb9eQQT_tg,Nam Em Official,KHÚC GIAO MÙA - NAM EM & NAM ANH | OFFICIAL MV...,NaN,NaN,10,vi,vi,...,True,public,False,1,https://en.wikipedia.org/wiki/Music|https://en...,"{""kind"": ""youtube#video"", ""etag"": ""sSGX7yy9nj0...",nhạc_tết_mv,/home/doanxem99/programming/school/DataVisuali...,nhạc tết mv,2026-04-30T16:26:07Z
4,Hc0fn7Z4Yms,2025-12-31T12:02:38Z,UCgLtSn3RGfd2-Qv6vbFTeig,Phê Music,Chưa Từng Đậm Sâu Remix - Inso (Bản Hít) | Anh...,NaN,NaN,10,vi,vi,...,True,public,False,27,https://en.wikipedia.org/wiki/Electronic_music...,"{""kind"": ""youtube#video"", ""etag"": ""YetjokFRD18...",nhạc_trẻ_music_video,/home/doanxem99/programming/school/DataVisuali...,nhạc trẻ music video,2026-04-30T16:01:38Z


## Tổng quan đặc trưng (features)

Bộ dữ liệu chứa 7353 mẫu video với 31 đặc trưng:

- `video_id` - Mã định danh duy nhất của video trên YouTube.
- `published_at` - Thời điểm video được đăng lên.
- `channel_id` - Mã định danh duy nhất của kênh đăng tải video.
- `channel_title` - Tên kênh đăng tải video.
- `title` - Tiêu đề đầy đủ của video.
- `description` - Nội dung mô tả chi tiết bên dưới video.
- `tags` - Danh sách các từ khóa/thẻ gắn kèm video.
- `category_id` - Mã số danh mục của video.
- `default_language` - Ngôn ngữ mặc định của video.
- `default_audio_language` - Ngôn ngữ mặc định của âm thanh video.
- `is_vietnamese_language_flag` - Cờ để kiểm tra video dùng tiếng Việt (có/không).
- `has_vietnamese_diacritics` - Mã vùng của video (vi/null).
- `in_publish_range` - Thời điểm tải lên nằm trong khoảng nhóm khảo sát (có/không).
- `accepted` - Thỏa mãn điều kiện (có/không) - mấy biến này mai t bỏ hếc.
- `filter_reason` - Hết bíc
- `view_count` - Tổng số lượt xem của video.
- `like_count` - Tổng số lượt thích của video.
- `comment_count` - Tổng số lượt bình luận của video.
- `duration` - Thời lượng video (theo chuẩn ISO 8601).
- `definition` - Chịu hua (chất lượng video hả).
- `caption` - Trạng thái video có phụ đề hay không (có/không).
- `licensed_content` - Tình trạng bản quyền của video (đúng ko z) (có/không).
- `privacy_status` - Chế độ riêng tư của video (private/unlisted/public).
- `made_for_kids` - Nội dung cho trẻ em (có/không).
- `tags_count` - Chi má.
- `topic_categories` - Các chủ đề liên quan (dạng link Wikipedia).
- `raw_json` - ?
- `source_folder` - bỏ nga.
- `source_csv_path` - như trên
- `search_query_used` - Truy vấn dùng để tìm video.
- `crawl_timestamp_utc` - Thời gian thu thập dữ liệu (theo chuẩn UTC).